### PR TITLE
Add info function

### LINKED ISSUE #23: Feature request: Print classifier info
It would be nice if there were a function that could show some info about the classifier, e.g. the expexted dimensionality of input data, the used features, etc.

### PR DESCRIPTION
**Pull request by jo-mueller**
This PR adds a `classifier.info()` function that displays all information in the header of the classifier. 

I added this to the `demo_pixel_classifier_multi_channel` along with the error message that comes up if dimensions do not match.

Fixes #23

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
This looks nice @jo-mueller ! It might be worth considering to use the `info()` function to also generate a nice `__str__` and/or `__repr__` to the class. That way when people do `print(classifier)` or `>> classifier` they get something that is understandable.

https://medium.com/swlh/understanding-repr-and-str-in-python-65dd41538943

File: apoc/_pixel_classifier.py [hunk 1]
```diff
@@ -103,6 +118,10 @@ def predict(self, image=None, features=None):
         if features is None:
             features = self.feature_specification
 
+        # if isinstance(image, np.ndarray):
```
Comment (haesleinhuepf) [inline apoc/_pixel_classifier.py:121]: Is this change necessary?

File: apoc/_pixel_classifier.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from sklearn.ensemble import RandomForestClassifier
 from ._converter import RFC_to_OCL
 from ._utils import generate_feature_stack, _read_something_from_opencl_file
+import re
```
Comment (haesleinhuepf) [inline apoc/_pixel_classifier.py:6]: Is this change necessary?

File: tests/test_multichannel_training_and_prediction.py [hunk 3]
```diff
@@ -38,3 +39,7 @@ def test_multichannel_training_and_prediction():
 
     assert np.allclose(result, ref_image)
 
+    clf.info()
+
+if __name__ == '__main__':
```
Comment (haesleinhuepf) [inline tests/test_multichannel_training_and_prediction.py:44]: Is this change necessary? Can't you run tests with `pytest`?

File: tests/test_training_on_folders.py [hunk 4]
```diff
@@ -34,4 +34,7 @@ def test_training_on_folders():
         labels = segmenter.predict(image)
 
         # There are at least 4 objects in each image
-        assert(labels.max() > 3)
\ No newline at end of file
+        assert(labels.max() > 3)
+
+if __name__ == '__main__':
```
Comment (haesleinhuepf) [inline tests/test_training_on_folders.py:39]: Is this change necessary?

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi Johannes @jo-mueller ,

can you please check if all the changes you did are necessary for the info() implementation? I also don't see why the notebooks were updated. I strongly suggest to send a new clean pull-request with only one commit adding this function and one new notebook showing how to use it.

Thanks!

Best,
Robert

**Comment by jo-mueller** [comment]
Hi Robert @haesleinhuepf , 

thanks for the feedback. 

> I also don't see why the notebooks were updated.

I found that the filenames for some notebooks were not very descriptive on their content (e.g., multi/single-channel prediction) so that I changed their names. To make it a bit easier to navigate the `demo` folder. I can of course, revert these changes if they're undesirable to you.

> I strongly suggest to send a new clean pull-request with only one commit adding this function and one new notebook showing how to use it.

Can do. That's rebasing/cherry-picking, right?

Also @kevinyamauchi , thanks for the idea, I'll give that a try :)

**Comment by haesleinhuepf** [comment]
> I found that the filenames for some notebooks were not very descriptive on their content (e.g., multi/single-channel prediction) so that I changed their names to make it a bit easier to navigate the `demo` folder.

Can you please creat an issue for this? This is obviously unrelated to the info() function. Thanks!

> That's rebasing/cherry-picking, right?

I would start from master with a clean branch and copy over the 10 lines of info() code. This appears easier to me that fit command wrangling. It's matter of taste obviously.

Thanks again!
